cpp/src/arrow/compute/kernels/CMakeLists.txt

```diff
@@ -18,10 +18,11 @@
 # ----------------------------------------------------------------------
 # Scalar kernels
 
+add_arrow_compute_test(scalar_cast_test SOURCES scalar_cast_test.cc test_util.cc)
```

Comment:
Why did you peel this off into its own test?

Reply:
It was so that the casts could be tested in isolation with/without the complete registry, given that none of the other scalar-type kernels are present in the base set. Also, the cast kernels seemed logically distinct enough to warrant a separate test IMO.

The testing situation isn't exactly ideal at the moment... The following will fail if ARROW_COMPUTE isn't enabled:
```
arrow-compute-scalar-type-test
arrow-compute-scalar-if-else-test
arrow-compute-scalar-temporal-test
arrow-compute-scalar-math-test
arrow-compute-scalar-utility-test
arrow-compute-vector-test
arrow-compute-aggregate-test
arrow-compute-expression-test
arrow-compute-plan-test
arrow-compute-hash-join-node-test
arrow-compute-asof-join-node-test
arrow-compute-groupby-test
```
I went into a bit more detail in an [earlier comment](https://github.com/apache/arrow/pull/34295#issuecomment-1441138674).

Reply:
Ah, ok.  That makes sense.